### DISCUSSION TITLE
[Issue]: Tricky wildcard names resolution in tuples deconstruction

### DISCUSSION THREAD

**Discussion by silver-sd** [dotnet/csharplang#4466] [General]
Current Implementation:
 ```
 (var1, _) = (1, 2); //treated as the wildcard if the variable '_' is not declared

  int _;
  //...
  (var1, _) = (1, 2); //changes '_' variable. This is a potential bug(unintentional change).
```

 Proposal:
   Always treat this as the wildcard even if the variable '_' is declared:

>  (var1, _) = (1, 2)

   If it is impossible(breaking change), at least generate a warning if the variable '_' is already declared. To reduce the probability of unintentional changes.

Like the current implementation of wildcards in 'out' parameters:
```
byte _ = 0;
F(out int _, out string _); //are always wildcards even if '_' is declared
```
The wildcard in 'switch' does not create any problems for using '_' as a variable too.

**Comment by svick** [comment]
It would be a breaking change, but it seems it's already being considered: https://github.com/dotnet/csharplang/issues/4460.
